File: 3069-distribute-elements-into-two-arrays-i/3069-distribute-elements-into-two-arrays-i.cpp
class Solution {
public:
    vector<int> resultArray(vector<int>& nums) {

        int n = nums.size();

        if (n <= 2) {
            return nums;
        }


        vector<int> arr1;
        vector<int> arr2;

        arr1.push_back(nums[0]);
        arr2.push_back(nums[1]);

        int ind = 2;

        while (ind < n) {
            if (arr1.back() > arr2.back()) {
                arr1.push_back(nums[ind]);
                ind++;
            } else {
                arr2.push_back(nums[ind]);
                ind++;
            }
        }

        for (int i : arr2) {
            arr1.push_back(i);
        }

        return arr1;
    }
};

File: 3069-distribute-elements-into-two-arrays-i/3069-distribute-elements-into-two-arrays-i.py
class Solution:
    def resultArray(self, nums: List[int]) -> List[int]:
        if len(nums) <= 2:
            return nums

        arr1 = [nums[0]]
        arr2 = [nums[1]]

        ind = 2

        while ind < len(nums):
            if arr1[-1] > arr2[-1]:
                arr1.append(nums[ind])
                ind+=1
            else:
                arr2.append(nums[ind])
                ind+=1
        
        res = []

        for i in arr1:
            res.append(i)

        for i in arr2:
            res.append(i)

        return res
